File: LLD/src/main/java/org/example/p1_oops/c1_accessmodifiers/p1/SamePackageNeighbor.java
package org.example.p1_oops.c1_accessmodifiers.p1;

public class SamePackageNeighbor {
    public void checkAccess() {
        System.out.println("\n--- Inside SamePackageNeighbor (Same Package, Different Class) ---");
        Parent p = new Parent();

        System.out.println("publicVar: " + p.publicVar + " (Accessible)");
        System.out.println("protectedVar: " + p.protectedVar + " (Accessible)");
        System.out.println("defaultVar: " + p.defaultVar + " (Accessible)");

        // System.out.println("privateVar: " + p.privateVar);
        System.out.println("privateVar: NOT Accessible (Private is only visible within the same class)");
    }
}

File: LLD NodeJs/p1_oops/c1_accessmodifiers/p1/SamePackageNeighbor.js
const Parent = require('./Parent');

/**
 * Demonstrates access within the same module/package
 * In JS, "same package" = same directory/module
 */
class SamePackageNeighbor {
    checkAccess() {
        console.log("\n--- Inside SamePackageNeighbor (Same Package, Different Class) ---");
        const p = new Parent();

        console.log("publicVar: " + p.publicVar + " (Accessible)");
        console.log("protectedVar: " + p._protectedVar + " (Accessible - convention only)");
        console.log("defaultVar: " + p.defaultVar + " (Accessible)");

        // console.log("privateVar: " + p.#privateVar);
        // SyntaxError: Private field '#privateVar' must be declared in an enclosing class
        console.log("privateVar: NOT Accessible (Private fields with # are only visible within the class)");
    }
}

module.exports = SamePackageNeighbor;
